.github/workflows/static-checks.yml

```diff
@@ -49,20 +49,28 @@ jobs:
     steps:
       - name: Checkout Code
         uses: actions/checkout@v2
-      - name: Setup PHP
+      - name: Setup PHP, with composer and extensions
         uses: shivammathur/setup-php@v2 #https://github.com/shivammathur/setup-php
         with:
           php-version: 7.4
+          extensions: mbstring, iconv, intl, mysql, zip
+          tools: composer
       - name: Setup Node
         uses: actions/setup-node@v2
         with:
           node-version: '14'
+      - name: Lint PHP code
+        run: cd SETUP && make lint_code && cd ..
       - name: Install NPM packages (for eslint)
         run: npm install
-      - name: Lint code
-        run: cd SETUP && make lint_code && cd ..
-      - name: Run js lint
+      - name: Run JS lint
         run: npm run lint
+      - name: Install composer dependencies
+        run: composer install
+      - name: Run PHP-CS-Fixer and confirm no changes
+        run: |
+          ./vendor/bin/php-cs-fixer fix
+          git diff --exit-code
```

Comment:
this seems kind of iffy to me. Could we not use some cli arg to php-cs-fixer, like --stop-on-violation
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.18/doc/usage.rst

Reply:
`--stop-on-violation` will stop after finding/fixing the first one but returns a zero exit code so the pipeline won't show a failure.

We could use `--dry-run --diff` which will return a non-zero exit code and still show the diff. I can change it to that.